java/vector/src/main/java/org/apache/arrow/vector/validate/ValidateVectorTypeVisitor.java

```diff
@@ -116,6 +116,25 @@ private void validateDateVector(ValueVector vector, DateUnit expectedDateUnit) {
         "Expecting date unit %s, actual date unit %s.", expectedDateUnit, dateType.getUnit());
   }
 
+  private void validateDecimalVector(ValueVector vector) {
+    validateOrThrow(vector.getField().getFieldType().getType() instanceof ArrowType.Decimal,
+            "Vector %s is not a decimal vector", vector.getClass());
+    ArrowType.Decimal decimalType = (ArrowType.Decimal) vector.getField().getFieldType().getType();
+    validateOrThrow(decimalType.getScale() >= 0, "The scale of decimal %s is negative.", decimalType.getScale());
```

Comment:
we should also validate precision is less then scale.

Reply:
Good point. Thanks. 
Added a check for that and a test case for it.